### DISCUSSION TITLE
(Not a bug) API returned 500, help needed to debug.

### DISCUSSION THREAD

**Discussion by kalyanoliveira** [abraunegg/onedrive#2757] [Q&A] [answered]
This is not a bug with the client, per se. Hence the reason I have not raised a bug report.

I keep getting an internal server error with the API:

```
➜  ~ onedrive --synchronize --download-only
Configuring Global Azure AD Endpoints

ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 500 ()
  Error Reason:     General Exception While Processing
  Error Timestamp:  2024-07-19T01:38:53
  API Request ID:   some numbers here which I am not sure I should disclose
```

I don't have any access to my 100 Gb stored in OneDrive, not even the Web UI is working. Can't create documents, can't see what files I have through the Web UI - it's just all gone for now.

Here's what the Web UI looks like, for reference:

![Thu Jul 18 10:47:32 PM -03 2024](https://github.com/user-attachments/assets/6a27b03d-9314-4dd5-bad2-b005bede4671)

This is how all of this happened:

> Today I needed to download all my files and upload them to an actual backup solution (restic), so I ran `onedrive --synchronize --download-only`. So good so far, my files were being downloaded one by one.

> I had left `onedrive --synchronize --download-only` running in a tmux session with `tmux new-session -s downloading onedrive --synchronize --download-only`. Because of the way tmux works, once the command given to `tmux new-session` is done running, the session is closed, and you sort of just loose history of the terminals that were inside it.

> It was quite slow (and I have a lot of files; 100 Gbs or so), so I just left it running and my laptop got locked (i3-lock). About 2 hours later, I unlocked my laptop, and I remember reading "download mis-match), but then it just resumed the download process.

> About half an hour later, I noticed that my tmux "dowloading" session, where I had left the client synching my files, was gone. "Good! My files must have finished downloading."

> Not really, though.
> ```
> ➜  ~ du -sh ./OneDrive
> 32G	./OneDrive
> ```
> Not the expected `100G`. "That's alright," I thought, "surely the client just encountered some sort of error, and running `onedrive --synchronize --download-only` will resume the downloading process from where it left of."

> That's when I got the same error message that I showed previously. I then tried opening the Web UI to see what happened, but it's as if I had no files. Creating new files also does not work, and it gives me a Correlation ID.

Yeah, I am sort of in the dark when it comes to debugging this. I've waited about 4 hours, and the issue still persists.

Is there any specific OneDrive API support channel where I can get talking with actual people to resolve the issue?

---

Also, I will say this. I know that OneDrive's API and the servers having some internal error (status 500) have nothing to do with this client, but I currently have no reason to believe that usage of the client was not what caused the issue. This was the first time that I attempted a large scale download of my OneDrive files with this client, I have previously only used `--single-dir`. Is there anything that I should be aware of, such as "do not perform large scale downloads, things will break," or something along the lines of that?

**Comment by kalyanoliveira** [comment]
I was reading your response [here](https://github.com/abraunegg/onedrive/issues/2666#issuecomment-2002126994), and saw that you talked about needing a full debug log in reference to an API status code 502. I know mine is actually 500, but anyways, here's as full log.

```log
[DEBUG] homePath: HOME environment variable set
[DEBUG] homePath: /home/kalyan
[DEBUG] configDirBase: WARNING - no XDG_CONFIG_HOME environment variable set
[DEBUG] configDirBase: /home/kalyan/.config
[DEBUG] configDirName: Configuring application to use default config path
Using 'user' Config Dir: /home/kalyan/.config/onedrive
Using 'system' Config Dir: /etc/onedrive
[DEBUG] refreshTokenFilePath = /home/kalyan/.config/onedrive/refresh_token
[DEBUG] deltaLinkFilePath = /home/kalyan/.config/onedrive/delta_link
[DEBUG] databaseFilePath = /home/kalyan/.config/onedrive/items.sqlite3
[DEBUG] databaseFilePathDryRun = /home/kalyan/.config/onedrive/items-dryrun.sqlite3
[DEBUG] uploadStateFilePath = /home/kalyan/.config/onedrive/resume_upload
[DEBUG] userConfigFilePath = /home/kalyan/.config/onedrive/config
[DEBUG] syncListFilePath = /home/kalyan/.config/onedrive/sync_list
[DEBUG] systemConfigFilePath = /etc/onedrive/config
[DEBUG] businessSharedFolderFilePath = /home/kalyan/.config/onedrive/business_shared_folders
No user or system config file found, using application defaults
[DEBUG] passed in options: ["onedrive", "--synchronize", "--download-only"]
[DEBUG] note --confdir and --verbose not listed in args
[DEBUG] sync_dir: Getting syncDir from config value sync_dir
[DEBUG] sync_dir: A '~' was found in configured sync_dir, automatically expanding as SHELL and USER environment variable is set
[DEBUG] syncDir: /home/kalyan/OneDrive
[DEBUG] Testing network to ensure network connectivity to Microsoft OneDrive Service
[DEBUG] Attempting to contact online service
[DEBUG] Shutting down HTTP engine as successfully reached OneDrive Online Service
Checking Application Version ...
[DEBUG] applicationVersion:       2.4.25
[DEBUG] latestVersion:            2.4.25
[DEBUG] publishedDate:            2023-Jun-21 05:30:55Z
[DEBUG] currentTime:              2024-Jul-19 01:29:59Z
[DEBUG] releaseGracePeriod:       2023-Jul-21 05:30:55Z
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
[DEBUG] Configured authUrl:          https://login.microsoftonline.com/common/oauth2/v2.0/authorize
[DEBUG] Configured redirectUrl:      https://login.microsoftonline.com/common/oauth2/nativeclient
[DEBUG] Configured tokenUrl:         https://login.microsoftonline.com/common/oauth2/v2.0/token
[DEBUG] Configured driveUrl:         https://graph.microsoft.com/v1.0/me/drive
[DEBUG] Configured driveByIdUrl:     https://graph.microsoft.com/v1.0/drives/
[DEBUG] Configured sharedWithMeUrl:  https://graph.microsoft.com/v1.0/me/drive/sharedWithMe
[DEBUG] Configured itemByIdUrl:      https://graph.microsoft.com/v1.0/me/drive/items/
[DEBUG] Configured itemByPathUrl:    https://graph.microsoft.com/v1.0/me/drive/root:/
[DEBUG] Configured siteSearchUrl:    https://graph.microsoft.com/v1.0/sites?search
[DEBUG] Configured siteDriveUrl:     https://graph.microsoft.com/v1.0/sites/
Using Curl defaults for all HTTP operations
[DEBUG] clientId    = not sure if I can share this number
[DEBUG] companyName = abraunegg
[DEBUG] appTitle    = OneDrive Client for Linux
Opening the item database ...
[DEBUG] Using database file: /home/kalyan/.config/onedrive/items.sqlite3
[DEBUG] Configuring default new folder permissions as: 700
[DEBUG] Configuring default new file permissions as: 600
All operations will be performed in: /home/kalyan/OneDrive
[DEBUG] Configuring skip_dir ...
[DEBUG] skip_dir: 
[DEBUG] Configuring skip_dir_strict_match ...
[DEBUG] skip_dir_strict_match: false
[DEBUG] Configuring skip_dotfiles ...
[DEBUG] skip_dotfiles: false
[DEBUG] Configuring skip_file ...
[DEBUG] skip_file: ~*|.~*|*.tmp
[DEBUG] Setting file permissions for: /home/kalyan/.config/onedrive/refresh_token
[DEBUG] Request URL = https://graph.microsoft.com/v1.0/me/drive
[DEBUG] oneDriveDetails	= onedrive.getDefaultDrive() generated a OneDriveException

ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 504 ()
[DEBUG] Raw Error Data: HTTP request returned status code 504 ()
{
    "error": {
        "code": "UnknownError",
        "innerError": {
            "client-request-id": "not sure if I can share this number either",
            "date": "2024-07-19T01:30:30",
            "request-id": "nor this one"
        },
        "message": ""
    }
}
[DEBUG] JSON Message: {"error":{"code":"UnknownError","innerError":{"client-request-id":"or this one too","date":"2024-07-19T01:30:30","request-id":"and perhaps maybe this one too"},"message":""}}
  Error Reason:     
  Error Timestamp:  2024-07-19T01:30:30
  API Request ID:   or this one
  Calling Function: init()


```

Not that promising tbh

**Comment by kalyanoliveira** [comment]
Well [this](https://portal.office.com/servicestatus) is alleviating. Seems there's currently an outage. Users are reporting things [here](https://learn.microsoft.com/en-us/answers/questions/1824209/one-driver-presenting-error-general-exception-whil) too. It seems that I was a victim of bad timing with the outage and me performing a large-scale sync.

**Comment by abraunegg** [comment, marked as answer]
@kalyanoliveira 
This page is always your friend when any client is generating 5xx errors: https://portal.office.com/servicestatus

As you know, any 5xx error message - regardless is a Server Side Issue and not a client side problem.

**Reply by kalyanoliveira** [reply]
Yep, this was my first time encountering this stuff. Thanks for the quick reply, everything is working once again.
